includes/rest-api/class-sensei-rest-api-extensions-controller.php

```diff
@@ -133,13 +165,125 @@ function( $plugin ) use ( $params ) {
+		$sensei_plugins = Sensei_Extensions::instance()->get_extensions( 'plugin' );
+
+		$plugins_to_update = array_filter(
+			$sensei_plugins,
+			function( $plugin ) use ( $plugins_arg ) {
+				return $plugin->is_installed && $plugin->has_update && in_array( $plugin->product_slug, $plugins_arg, true );
+			}
+		);
+
+		if ( empty( $plugins_to_update ) ) {
+			$response = new WP_REST_Response();
+			$response->set_data(
+				new WP_Error(
+					'sensei_extensions_no_plugins_to_update',
+					__( 'No plugins to update found.', 'sensei-lms' )
+				)
+			);
+			$response->set_status( 404 );
+
+			return $response;
+		}
+
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+		require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+
+		WP_Filesystem();
+
+		$skin     = new WP_Ajax_Upgrader_Skin();
+		$upgrader = new Plugin_Upgrader( $skin );
+		$result   = $upgrader->bulk_upgrade( wp_list_pluck( $plugins_to_update, 'plugin_file' ) );
+
+		$error = $this->check_for_upgrade_error( $plugins_to_update, $result, $skin, $upgrader );
+
+		if ( is_wp_error( $error ) ) {
+			return $error;
+		}
+
+		return $this->create_plugins_response( Sensei_Extensions::instance()->get_extensions( 'plugin' ) );
+	}
+
+	/**
+	 * Check if the result of the upgrade has an error. Error handling has been copied from wp_ajax_update_plugin.
+	 *
+	 * @param array                 $plugins  Plugins which where upgraded.
+	 * @param array|WP_Error|false  $result   Result of the upgrade.
+	 * @param WP_Ajax_Upgrader_Skin $skin     Upgrader sking.
+	 * @param Plugin_Upgrader       $upgrader The upgrader.
+	 *
+	 * @return bool|WP_Error
+	 */
+	private function check_for_upgrade_error( array $plugins, $result, WP_Ajax_Upgrader_Skin $skin, Plugin_Upgrader $upgrader ) {
+		if ( is_wp_error( $skin->result ) ) {
+			return $skin->result;
+		}
+
+		if ( $skin->get_errors()->has_errors() ) {
+			return new WP_Error(
+				'sensei_extensions_plugin_update_failed',
+				$skin->get_error_messages()
+			);
+		}
+
+		if ( is_array( $result ) ) {
+			foreach ( $plugins as $plugin ) {
+				if ( empty( $result[ $plugin->plugin_file ] ) ) {
+					return new WP_Error(
+						'sensei_extensions_plugin_update_failed',
+						// translators: Placeholder is the name of the plugin that failed.
+						sprintf( __( 'Failed to update plugin %s', 'sensei-lms' ), $plugin->title )
+					);
+				}
+
+				if ( true === $result[ $plugin->plugin_file ] ) {
+					return new WP_Error(
+						'sensei_extensions_plugin_update_failed',
+						$upgrader->strings['up_to_date']
+					);
+				}
+			}
+		} else {
+			return new WP_Error(
+				'sensei_extensions_plugin_update_failed',
+				__( 'Plugin update failed.', 'sensei-lms' )
+			);
+		}
+
+		return false;
+	}
+
+	/**
+	 * Generate a REST response from an array of plugins.
+	 *
+	 * @param array $plugins The plugins.
+	 *
+	 * @return WP_REST_Response
+	 */
+	private function create_plugins_response( array $plugins ): WP_REST_Response {
```

Comment:
Would make more sense to call it `create_extensions_response`?

Reply:
Fixed in 72d93de64dc33988741faaf2ad76689616ce372f